Fixed a crash when going from Media > Album view

calvinnor opened the pull request:
Bugs:
- When a user goes from Album > Media view, then minimizes the
  application and the activity gets destroyed in background,
  the resumed activity crashes when pressing back.
- When a user minimizes the application and the activity gets
  destroyed, the resumed app has non-working Navigation Drawer.

Fixes:
- Save 'albumsMode' state in onSaveInstanceState() so that
  we know if the destroyed activity was in Album / Media mode.
- Initialised the UI and set onClickListeners for Navigation
  Drawer irrespective of savedInstanceState.

Quirks:
- The activity does not crash and navigates correctly to previous
  screen, however the media does not show up when resuming.

calvinnor committed:
Fixed a crash when going from Media > Album view

Bugs:
- When a user goes from Album > Media view, then minimizes the
  application and the activity gets destroyed in background,
  the resumed activity crashes when pressing back.
- When a user minimizes the application and the activity gets
  destroyed, the resumed app has non-working Navigation Drawer.

Fixes:
- Save 'albumsMode' state in onSaveInstanceState() so that
  we know if the destroyed activity was in Album / Media mode.
- Initialised the UI and set onClickListeners for Navigation
  Drawer irrespective of savedInstanceState.

Quirks:
- The activity does not crash and navigates correctly to previous
  screen, however the media does not show up when resuming.

dnldsht commented:
Do you know any trick to get the main Activity destroyed while in the background?

CalvinNor commented:
@dnldsht Android has a setting called "Don't Keep Activities" which simulates a low-memory device.

What it does is kills the activity as soon as it's put into background, so that you can test the restoring of state on old devices.

Find it under Developer Options > Last Section "Apps"

dnldsht committed:
media does show up when resuming

dnldsht committed:
clean RvMediaFragment

dnldsht committed:
- created listeners for EditMode and nothing to show
- removed MainActivity dependency from AlbumFragments and RvMediaFragments
- added click listener for albums
